packages/compass-indexes/src/modules/create-index.tsx

```diff
@@ -866,11 +854,7 @@ const reducer: Reducer<State, Action> = (state = INITIAL_STATE, action) => {
   ) {
     return {
       ...state,
-      coveredQueriesObj: {
-        coveredQueries: action.coveredQueries,
-        optimalQueries: action.optimalQueries,
-        showCoveredQueries: action.showCoveredQueries,
-      },
+      coveredQueriesArr: action.coveredQueriesArr,
```

Comment:
Already better, but generally speaking reducer when possible should be responsible for the state shape. Basically your case here is user clicked a button (and so the action is actually something more like "show covered queries clicked"), and the state can be completely calculated in the reducer

```suggestion
      coveredQueriesArr: state.fields.map((field, index) => {
        return { [field.name]: index + 1 };
      }),
```